sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -600,16 +601,23 @@ class GpuSpecifiedWindowFrameMeta(
           case Literal(value, ShortType) => value.asInstanceOf[Short].toLong
           case Literal(value, IntegerType) => value.asInstanceOf[Int].toLong
           case Literal(value, LongType) => value.asInstanceOf[Long]
-          case Literal(value, CalendarIntervalType) =>
-            val interval = value.asInstanceOf[CalendarInterval]
-            if (interval.microseconds != 0 || interval.months != 0) { // DAYS == 0 is permitted.
-              willNotWorkOnGpu(s"Bounds for Range-based window frames must be specified" +
-                s" only in DAYS. Found $interval")
+          case Literal(ci: CalendarInterval, CalendarIntervalType) =>
+            if (ci.months != 0) {
+              willNotWorkOnGpu("interval months isn't supported")
+            }
+            // return the total microseconds
+            try {
+              Math.addExact(
+                Math.multiplyExact(ci.days.toLong, TimeUnit.DAYS.toMicros(1)),
+                ci.microseconds)
+            } catch {
+              case e: ArithmeticException =>
+                willNotWorkOnGpu("interval total microseconds has overflow")
```

Comment:
I would like to see a better message. It is not clear how the user can fix this or what exactly has happened.

Perhaps something more like

```
willNotWorkOnGpu(s"windows over timestamps are converted to microseconds and $ci is too large to fit")
```

Reply:
Thx @revans2. Done